File: enum/3047求交集区域内的最大正方形面积.cpp
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
public:
    long long largestSquareArea(vector<vector<int>>& bottomLeft, vector<vector<int>>& topRight) {
        int res = 0;
        int n = bottomLeft.size();
        for(int i = 0; i < n; i++){
            int bx = bottomLeft[i][0];
            int by = bottomLeft[i][1];
            int tx = topRight[i][0];
            int ty = topRight[i][1];
            for(int j = 0; j < i; j++){
                int bx2 = bottomLeft[j][0];
                int by2 = bottomLeft[j][1];
                int tx2 = topRight[j][0];
                int ty2 = topRight[j][1];
                int width = min(tx,tx2)-max(bx,bx2);
                int weight = min(ty,ty2)-max(by,by2);
                res = max(res,min(width,weight));
            }
        }
        return 1LL*res*res;
    }
};

File: enum/3047求交集区域内的最大正方形面积.py
from typing import List
class Solution:
    def largestSquareArea(self, bottomLeft: List[List[int]], topRight: List[List[int]]) -> int:
        res = 0
        for i,((bx,by),(tx,ty)) in enumerate(zip(bottomLeft,topRight)):
            for j in range(i):
                bx2, by2 = bottomLeft[j]
                tx2, ty2 = topRight[j]
                width = min(tx, tx2) - max(bx, bx2)
                height = min(ty, ty2) - max(by, by2)
                res = max(res, min(width, height))
        return res ** 2